pandas/tests/indexing/test_floats.py

```diff
@@ -206,9 +207,9 @@ def test_scalar_integer(self):
         # test how scalar float indexers work on int indexes
 
         # integer index
-        for index in [tm.makeIntIndex, tm.makeRangeIndex]:
```

Comment:
rather than change this, wouldn't it be better to simply fix the ``make*`` ones? this is pretty much an anti-patter and the reason we have the helpers.

Reply:
The only way to "fix" ``makeIntIndex`` for this test to work is to leave it as it is, but as I stated, it limits comparability across index types, since others just return random stuff, i.e. unsorted.

This test does something different than using the ``makeIntIndex`` to get "some index filled with integers": it assumes that the result has a specific content. I see _this_ as an anti-pattern.

Reply:
why are you removing testing of RangeIndex? (yes you convert to it, but its not very explicit)

Reply:
It's just simpler, but OK, I can make it explicit

Reply:
(done), ping